src/WebJobs.Extensions.DurableTask/LinuxAppServiceFileLogger.cs

```diff
@@ -173,9 +170,14 @@ private void RollFiles()
             // Rename current file to older file.
 
 #if !FUNCTIONS_V1
-            Syscall.rename(this.logFilePath, this.archiveFilePath);
+            rename(this.logFilePath, this.archiveFilePath);
 #endif
 
         }
+
+#if !FUNCTIONS_V1
+        [DllImport("libc", SetLastError = true)]
```

Comment:
How does `[DllImport]` work if the library is not available (for instance, on Windows)?

Does this dynamically try to import this functionality when `rename` is actually called? In that case, this likely isn't an issue since this logger should never actually be used on Windows.

Reply:
My understanding is that `DllImport` will not actually seek to invoke any system-level libraries until `rename` is called, so this shouldn't be an issue.

From this SO answer: https://stackoverflow.com/questions/39144458/what-happens-if-i-have-dllimport-c-net-for-an-api-not-supported-by-the-opera

Reply:
In that case, we should decide on whether in the degenerate case this logger class is used on a non-Linux platform if we fail fast with an exception, or we should wrap the call to `rename()` in a try-catch.

@cgillum, what are your thoughts in this scenario? I normally lean towards just letting the exception propagate, but since this is logging code I'm never quite sure of proper error handling.

Reply:
I +1 failing fast if we're somehow using the logger in windows

Reply:
Don't we have code that always ensures we're only running this logger in Linux? I'm fine with letting it fail on Windows, though I recommend we log the failure before rethrowing any exception so that someone debugging will have a bit more context.

Reply:
So the suggestion is to: catch, then log, then re-throw? Got it :)